### PR TITLE
Add GH action CI job to enable CodeCov

### PR DESCRIPTION
**Pull request by pankajastro**
Enable Code Coverage for the repo
Add CI job for Code Coverage
Post this change CodeCov will post a comment about Code Cov status on the PR

### CODE REVIEW

File: anyscale_provider/operators/anyscale.py [hunk 1]
```diff
@@ -329,6 +329,10 @@ def on_kill(self) -> None:
             self.log.info("Termination request received. Submitted request to terminate the anyscale service rollout.")
         return
 
+    @staticmethod
+    def test_me() -> None:
+        assert 1 == 1
+
```
Comment (pankajastro) [inline anyscale_provider/operators/anyscale.py:335]:
```suggestion
    @staticmethod
    def test_me() -> None:
        assert 1 == 1

```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ name: test
 
 on:
   push:
-    branches: [ "main" ]
+    branches: [ "main", "test_codecov" ]
```
Comment (pankajastro) [inline .github/workflows/test.yml:5]:
```suggestion
    branches: [ "main"  ]
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ on:
   push:
     branches: [ "main" ]
   pull_request:
-    branches: [ "main" ]
+    branches: [ "main", "test_codecov" ]
```
Comment (pankajastro) [inline .github/workflows/test.yml:7]:
```suggestion
    branches: [ "main" ]
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -111,3 +111,36 @@ jobs:
...
+        uses: actions/download-artifact@v2
+        with:
+          path: ./coverage
+      - name: Combine coverage
+        run: |
+          coverage combine ./coverage/coverage*/.coverage
+          coverage report
+          coverage xml
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```
Comment (pankajkoti) [inline .github/workflows/test.yml:142]:
```suggestion
        uses: codecov/codecov-action@v4
```
should we try v4?

**Comment by pankajastro** [comment]
Look like code cov will start working once this changes will merged in default branch 
> Once merged to your default branch, subsequent pull requests will have Codecov checks and comments. Additionally, you’ll find your repo coverage dashboard here. If you have merged, try reloading the page.

**Review by pankajkoti — DISMISSED** [review]

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**